### ISSUE TITLE
Unified objects' secrets handling

### ISSUE THREAD

**Issue by vholer** [OpenNebula/one#3064]
**Description**
Today the secrets handling is split between 2 parts:
1. oned ensures the (host) secrets are encrypted on update
https://github.com/OpenNebula/one/blob/ea86bd2039ef30a5e3bdfcd7e3f295cac23f880e/src/host/Host.cc#L721-L746
2. (cloud) drivers decrypt the secrets on their own
https://github.com/OpenNebula/one/blob/ea86bd2039ef30a5e3bdfcd7e3f295cac23f880e/src/vmm_mad/remotes/ec2/ec2_driver.rb#L304-L310

With DDC, we have extended the number of components which need to access the secrets. For IPAM drivers, the secrets might be part of the AR or VNET and to decrypt them, they would have to replicate the same behaviour of VM drivers. But, for VNET drivers, since they are executed on the remote hosts, it's impossible to decrypt the secrets as they don't have access to the decryption key.

The idea is to:
1. handle all encryption / decryption of secrets only inside oned
2. have the list of secrets configurable (oned.conf)
3. for the drivers actions, decrypt the secrets in passed XMLs
4. ensure we don't pass the XMLs with unencrypted secrets as **arguments** and in case of failure, they aren't persisted in any log
5. detect encrypted secret a better way than by the size:
https://github.com/OpenNebula/one/blob/ea86bd2039ef30a5e3bdfcd7e3f295cac23f880e/src/host/Host.cc#L744
6. hooks (VM, HOST, VNET) might need access to the plain secrets as well; consider passing the object XML on stdin

Basically, the secrets will leave encrypted in regular API calls, but the drivers actions will get XMLs with decrypted secrets.

## Progress Status
- [x] Branch created 
- [x] Code committed to development branch
- [x] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by vholer** [comment]
A little related issue: https://github.com/OpenNebula/one/issues/1408

**Comment by paczerny** [comment]
PRs for merge:
* one: #3616 into master
* development: OpenNebula/development#698 into master
* docs: OpenNebula/docs#726 into master
